From 84e2fcc050cd751b96f03a67e8343222817a8646 Mon Sep 17 00:00:00 2001
From: Julian Bouzas <julian.bouzas@collabora.com>
Date: Thu, 26 Jun 2025 14:05:06 -0400
Subject: [PATCH] monitor/alsa: Increase priority for USB devices

We always want the plugged USB ALSA device to have higher priority than the
internal ALSA device.
---
 src/scripts/monitors/alsa.lua | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/scripts/monitors/alsa.lua b/src/scripts/monitors/alsa.lua
index e3cd45db..ce87ebaa 100644
--- a/src/scripts/monitors/alsa.lua
+++ b/src/scripts/monitors/alsa.lua
@@ -220,6 +220,10 @@ function createNode(parent, id, obj_type, factory, properties)
       priority = priority + 8
     end
 
+    if dev_props["device.bus"] == "usb" then
+      priority = priority + 100
+    end
+
     properties["priority.driver"] = priority
     properties["priority.session"] = priority
   end
-- 
GitLab

